core/types/src/keys.rs

```diff
@@ -117,10 +148,20 @@ impl<ADDR, KIND> TryFrom<&[u8; 32]> for Key<ADDR, KIND, RistrettoPublic> {
     }
 }
 
-/// Access underlying [`RistrettoPoint`] for public keys using [`AsRef`]
-impl<ADDR, KIND> AsRef<RistrettoPoint> for Key<ADDR, KIND, RistrettoPublic> {
-    fn as_ref(&self) -> &RistrettoPoint {
-        self.key.as_ref()
+/// Attempt to create a public key from a compressed point, wrapping
+/// [`RistrettoPublic::try_from`]
+impl<ADDR, KIND> TryFrom<[u8; 32]> for Key<ADDR, KIND, RistrettoPublic> {
+    type Error = KeyError;
+
+    fn try_from(p: [u8; 32]) -> Result<Self, Self::Error> {
+        Self::try_from(&p)
+    }
+}
+
+/// Access underlying [`RistrettoPoint`] for public key containers
+impl<ADDR, KIND> From<&Key<ADDR, KIND, RistrettoPublic>> for RistrettoPoint {
```

Comment:
from using this more, having multiple `AsRef` impls becomes _very_ unclear / confusing when converting types, i think `From` is more clear, though it does usually encompass ownership. opinions / preferences either way?